test/EFCore.PG.FunctionalTests/Query/EnumQueryTest.cs

```diff
@@ -38,14 +39,72 @@ public void Where_with_parameter()
         {
             using (var ctx = CreateContext())
             {
-                var sad = EnumType1.Sad;
-                var x = ctx.SomeEntities.Single(e => e.Enum1 == sad);
-                Assert.Equal(EnumType1.Sad, x.Enum1);
-                AssertContainsInSql("(DbType = Object)");  // Not very effective but better than nothing
-                AssertContainsInSql("WHERE e.enum1 = @");
+                // ReSharper disable once ConvertToConstant.Local
+                var sad = MappedEnum.Sad;
+                var x = ctx.SomeEntities.Single(e => e.MappedEnum == sad);
+                Assert.Equal(MappedEnum.Sad, x.MappedEnum);
+                AssertContainsInSql("(DbType = Object)"); // Not very effective but better than nothing
+                AssertContainsInSql("WHERE e.\"MappedEnum\" = @__sad_0");
+            }
+        }
+
+        [Fact]
+        public void Where_with_unmapped_enum_parameter_downcasts_are_implicit()
+        {
+            using (var ctx = CreateContext())
+            {
+                // ReSharper disable once ConvertToConstant.Local
+                var sad = UnmappedEnum.Sad;
+                var _ = ctx.SomeEntities.Single(e => e.UnmappedEnum == sad);
+                AssertContainsInSql("@__sad_0='?' (DbType = Int32)");
+                AssertContainsInSql("WHERE e.\"UnmappedEnum\" = @__sad_0");
+            }
+        }
+
+        [Fact]
+        public void Where_with_unmapped_enum_parameter_downcasts_do_not_matter()
+        {
+            using (var ctx = CreateContext())
+            {
+                // ReSharper disable once ConvertToConstant.Local
+                var sad = UnmappedEnum.Sad;
+                var _ = ctx.SomeEntities.Single(e => (int)e.UnmappedEnum == (int)sad);
+                AssertContainsInSql("@__sad_0='?' (DbType = Int32)");
+                AssertContainsInSql("WHERE e.\"UnmappedEnum\" = @__sad_0");
             }
         }
 
+        [Fact]
+        public void Where_with_mapped_enum_parameter_downcasts_do_not_matter()
+        {
+            using (var ctx = CreateContext())
+            {
+                // ReSharper disable once ConvertToConstant.Local
+                var sad = MappedEnum.Sad;
+                var _ = ctx.SomeEntities.Single(e => (int)e.MappedEnum == (int)sad);
+                AssertContainsInSql("@__sad_0='?' (DbType = Object)");
+                AssertContainsInSql("WHERE e.\"MappedEnum\" = @__sad_0");
+            }
+        }
+
+        [Fact]
+        public void Where_with_unmapped_enum_parameter_downcast_for_int_comparison_does_matter()
```

Comment:
Like I wrote on a previous I'm issue, I'm not sure of the value of having negative tests like this - but I'm OK with it.

Reply:
I figured it would help monitor changes in type handling at the provider-level, and parameter handling for EF Core in general. I'm also thinking about adding some info on this issue to the enum mapping docs that would link here for an example. The idea being that if something changes, this test will start to fail (in a good way!) and we can update the docs accordingly.

But I could also accomplish this with an untracked file in my fork. Let me know if you change your mind and want to omit it, I'm good either way.